include/alpaka/KernelBundle.hpp

```diff
@@ -100,14 +101,26 @@ namespace alpaka
         template<typename TAcc>
         requires(
             alpaka::concepts::KernelFn<KernelFn>
-            && std::same_as<
-                void,
-                std::invoke_result_t<
-                    KernelFn,
-                    TAcc,
-                    remove_restrict_t<ALPAKA_TYPEOF(onHost::makeAccessibleOnAcc(std::declval<TArgs>()))>...>>)
+            && std::is_invocable_v<
+                std::remove_const_t<KernelFn>,
+                TAcc,
+                remove_restrict_t<ALPAKA_TYPEOF(onHost::makeAccessibleOnAcc(std::declval<TArgs>()))>...>)
         constexpr auto operator()(TAcc const& acc) const
         {
+            static_assert(
+                std::is_invocable_v<
+                    std::add_const_t<KernelFn>,
```

Comment:
I saw this, after I approved :-( How is this working? It looks like you make the function const.

Reply:
The idea is to add const to the invocable check to check whether the operator is callable when the object is const, since it must be. It's not actually muting anything.

Reply:
I'm not sure what you mean, `KernelFn` is an object implementing `operator()`, or at least it should be. If that `operator()` is not marked const, it can't be called later on, and to give a better error message, this is now checked here. If it is marked const, the operator is callable on the constant `KernelFn` object and everything is fine.